tests/transforms/test_mitigate.py

```diff
@@ -208,6 +209,34 @@ def original_qnode(inputs):
         result_expanded = mitigated_qnode_expanded(inputs)
         assert qml.math.allclose(result_orig, result_expanded)
 
+    # pylint:disable=not-callable
+    def test_zne_with_noise_models(self):
+        """Test that mitigate_with_zne transform works with noise models"""
+        fcond = qml.noise.wires_in([0, 1])
+        noise = qml.noise.partial_wires(qml.AmplitudeDamping, 0.05)
+        noise_model = qml.NoiseModel({fcond: noise})
+
+        def circuit():
+            qml.RX(1.23, wires=0)
+            qml.RZ(0.45, wires=1)
+            return qml.expval(qml.Z(0) @ qml.Z(1))
+
+        with pytest.raises(
+            ValueError,
+            match="Circuits with channels cannot be folded with mitigate_with_zne.",
+        ):
+            noise_qnode = qml.add_noise(qml.QNode(circuit, device=dev_ideal), noise_model)
+            zne_qnode = qml.transforms.mitigate_with_zne(
+                noise_qnode, [1, 2, 3], fold_global, richardson_extrapolate
+            )
+            zne_qnode()
+
+        noise_qnode = qml.QNode(circuit, device=qml.add_noise(dev_ideal, noise_model))
+        zne_qnode = qml.transforms.mitigate_with_zne(
+            noise_qnode, [1, 2, 3], fold_global, richardson_extrapolate
+        )
+        assert qml.math.allclose(zne_qnode(), 0.3984378845598476, atol=1e-2)
```

Comment:
How has been calculated this value?

Reply:
I computed it manually via `mitiq`.

Reply:
@obliviateandsurrender do you have the source code for that? Just curious :)

Reply:
Please add some information that allows reproducing the reference value, ideally the full code or the important part of it if too long.

Reply:
I personally obtained it with some local changes to `mitiq` to extract the results of their `global_folding` functionality and then feed it to their `RichardsonFactory` method, while defining the circuit in `cirq` as below:
```py
circuit = cirq.Circuit()
q0, q1 = cirq.LineQubit.range(2)
circuit.append(cirq.rx(1.23).on(q0))
circuit.append(cirq.rz(0.45).on(q1))

z = np.diag([1, -1])
hamiltonian = np.kron(z, z)
noiseless_executor = lambda circ: np.real(
    np.trace(cirq.DensityMatrixSimulator().simulate(circ).final_density_matrix @ hamiltonian)
)
executor_with_noise = lambda circ: noiseless_executor(
    circ.with_noise(cirq.amplitude_damp(0.01))
)
```
An easier alternative could be to simply use mitiq's own function with our transform:
```py
>>> qml.transforms.mitigate_with_zne(
    noise_qnode, [1, 2, 3], 
    mitiq.zne.scaling.fold_global, mitiq.zne.inference.RichardsonFactory.extrapolate
)()
0.3984378845760577
```
I have added that in the comment.